packages/legend-application-studio/src/components/editor/editor-group/function-activator/__tests__/FunctionActivator.test.tsx

```diff
@@ -0,0 +1,76 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect, test } from '@jest/globals';
+import { MockedMonacoEditorInstance } from '@finos/legend-lego/code-editor/test';
+import { integrationTest } from '@finos/legend-shared/test';
+import { fireEvent, getByText, waitFor } from '@testing-library/react';
+import {
+  TEST__provideMockedEditorStore,
+  TEST__setUpEditorWithDefaultSDLCData,
+} from '../../../__test-utils__/EditorComponentTestUtils.js';
+import TEST_DATA__SimpleRelationalModel from './TEST_DATA__SimpleRelationalEntities.json' assert { type: 'json' };
+import { LEGEND_STUDIO_TEST_ID } from '../../../../../__lib__/LegendStudioTesting.js';
+import { Core_GraphManagerPreset } from '@finos/legend-graph';
+import { LegendStudioPluginManager } from '../../../../../application/LegendStudioPluginManager.js';
+
+const pluginManager = LegendStudioPluginManager.create();
+pluginManager.usePresets([new Core_GraphManagerPreset()]).install();
+
+test(integrationTest('Test Function Activator '), async () => {
+  const MOCK__editorStore = TEST__provideMockedEditorStore({
+    pluginManager,
+  });
+  const renderResult = await TEST__setUpEditorWithDefaultSDLCData(
+    MOCK__editorStore,
+    {
+      entities: TEST_DATA__SimpleRelationalModel,
+    },
+  );
+  MockedMonacoEditorInstance.getValue.mockReturnValue('');
+  MockedMonacoEditorInstance.getRawOptions.mockReturnValue({
+    readOnly: true,
+  });
+
+  const functionPackageName = 'model';
+  const functionName = 'Firm_QueryFunction():TabularDataSet[1]';
+
+  const explorerTree = renderResult.getByTestId(
+    LEGEND_STUDIO_TEST_ID.EXPLORER_TREES,
+  );
+  fireEvent.click(getByText(explorerTree, functionPackageName));
+  fireEvent.click(getByText(explorerTree, functionName));
+
+  const functionEditor = renderResult.getByTestId(
+    LEGEND_STUDIO_TEST_ID.FUNCTION_EDITOR,
+  );
+  fireEvent.click(getByText(functionEditor, 'Promote'));
+  const functionActivatorModal = await waitFor(() =>
+    renderResult.getByRole('dialog'),
+  );
+  expect(
+    getByText(functionActivatorModal, 'Snowflake Native App'),
+  ).toBeDefined();
+  fireEvent.click(getByText(functionActivatorModal, 'Snowflake Native App'));
+  expect(getByText(functionActivatorModal, 'Target Path')).toBeDefined();
+  fireEvent.click(getByText(functionActivatorModal, 'Promote'));
+  fireEvent.click(getByText(explorerTree, 'NewActivator'));
+
+  const editorGroupHeaders = renderResult.getByTestId(
+    LEGEND_STUDIO_TEST_ID.EDITOR_GROUP__HEADER_TABS,
+  );
+  expect(getByText(editorGroupHeaders, 'NewActivator')).toBeDefined();
```

Comment:
can we do checks ensure we have selected the first connection and see the function lable, name. etc.

Reply:
I bet you won't catch this 👀🫡
lol i tried and failed somehow it will trigger this API  (../external/format/availableFormats)
Then I tried to mock this API.. then the page is empty

Reply:
could we dig into this and why thats happening. that api should be triggered on the first launch of editor not during this flow